### DISCUSSION TITLE
Unclear how all the pieces fit together

### DISCUSSION THREAD

**Discussion by scottj97** [zwave-js/zwave-js#6237] [General] [answered]
I'm new to Z-Wave, though I have some home automation programming experience (Insteon modem). I'm trying to control my new Leviton Z-Wave dimmer/fan switches using my new Zooz ZST10 700 Z-Wave USB stick from my custom automation system written in Ruby.

Everybody recommends Home Assistant and its Z-Wave JS plugin, but it seems like overkill for what I want (a software API).

My goals: (1) query light/fan status at any time; (2) receive notifications when the state is manually changed; (3) control the state. Bonus: specify a ramp rate when I change the dimmer.

Can/should I skip Home Assistant and use only the zwave-js stuff? If yes can I use only the node-zwave-js repo and somehow control it from Ruby? Or maybe that's what the zwave-js-server is for? Do I also need MQTT, which seems to be in zwave-js-ui (which is weird because I don't think I need a UI)?

It's not clear to me how all the pieces fit together and which ones I really need. The [project README](https://github.com/zwave-js/.github/blob/master/profile/README.md) lists all these projects, but doesn't really explain how they all fit together, or when I would want/need to use each one.

**Comment by AlCalzone** [comment, marked as answer]
You can do what Home Assistant does internally:
Execute an instance of https://github.com/zwave-js/zwave-js-server, which wraps `node-zwave-js` and exposes the API via a websocket, then interact with the websocket from the Ruby side.

We try to keep the server component relatively up to date with `node-zwave-js`. If something is missing, just open an issue in the server repo.

**Reply by scottj97** [reply]
I will check that out.

[Elsewhere](https://www.avsforum.com/threads/z-wave-programming-confusion.3283203/#post-62793031) I received advice that I should use zwave-js-ui and its MQTT feature. Would that give me any additional capabilities beyond zwave-js-server?

Should the project README have a section explaining these kind of things? To answer these questions for each subproject:
* Who is this for?
* Why would I want to use this?
* What can this do for me?

**Reply by AlCalzone** [reply]
IMO, interfacing the API through MQTT is needlessly complicated. The Websocket API gives you a much more "natural" interface.
That said, Z-Wave JS UI also hosts the websocket server by default, so you can also run that for a feature-rich UI and tailor your project to do just what you need it to.

> Who is this for?
> Why would I want to use this?
> What can this do for me?

Happy to take some suggestions for this.
